### ISSUE TITLE
Some tests mutate topics

### ISSUE THREAD

**Issue by betalb** [crossfilter/crossfilter#90]
Some tests mutate original topic. This may lead to wrong test results if one of test assertions fail before doing  cleanup.

One of such tests &ndash; `remove with predicate`

Consider following example
```
"remove with predicate": {
  topic: function() {
    return crossfilter();
  },
  "test 1": function(data) {
    data.add([{foo: 1}, {foo: 1.1}, {foo: 1.2}]);

    assert.equal(data.size(), 3);
    
    data.remove(function () {
      return true;
    });
  },
  "test 2": function(data) {
    data.add([{foo: 1}, {foo: 1.1}, {foo: 1.2}]);

    assert.equal(data.size(), 3);

    data.remove(function () {
      return true;
    });
  },
  "test 3": function() {
    data.add([{foo: 1}, {foo: 1.1}, {foo: 1.2}]);

    assert.equal(1, 2);

    data.remove(function () {
      return true;
    });
  }
}
```

Hash iteration is not specified by ECMAScript and is implementation dependent. So above test may executed test 1, test 2, test 3 in any order:

Expected
```
test 1 passed
test 2 passed
test 3 failed
```

Not Expected
```
test 1 passed
test 3 failed
test 2 failed
```

Not Expected
```
test 3 failed
test 1 failed
test 2 failed
```

**Comment by betalb** [comment]
I may be able to update tests to recreate topic on every run

Or jasmine / mocha have better API to execute actions before every test / test suite

**Comment by esjewett** [comment]
This is a known issue. Thanks for filing it and getting it recorded. If you or anyone else are able to rewrite any or all of these tests to work independently, it would be much appreciated. I would probably recommend creating a re-usable setup/teardown method that runs for each test, either using Jasmine/Mocha APIs or another way. Please keep in mind that many of the current tests *expect* the previous tests to have run and so test expectations will need to be carefully updated for many tests.

**Comment by gordonwoodhull** [comment]
Fixed by #176
